src/Storages/MergeTree/MergeTreeIndexText.cpp

```diff
@@ -516,10 +516,20 @@ bool MergeTreeIndexGranuleText::hasAnyTokenFromQuery(const TextSearchQuery & que
         if (remaining_tokens.contains(token))
             return true;
     }
-    return query.tokens.empty();
+    return false;
 }
 
 bool MergeTreeIndexGranuleText::hasAllTokensFromQuery(const TextSearchQuery & query) const
+{
+    for (const auto & token : query.tokens)
+    {
```

Comment:
Minor: the braces here are not needed (personally I prefer them, but just mention).